sda-download/api/sda/sda.go

```diff
@@ -166,6 +167,60 @@ func Download(c *gin.Context) {
 		return
 	}
 
+	// Get query params
+	qStart := c.DefaultQuery("startCoordinate", "0")
+	qEnd := c.DefaultQuery("endCoordinate", "0")
+
+	// Parse and verify coordinates are valid
+	start, err := strconv.ParseInt(qStart, 10, 0)
+
+	if err != nil {
+		log.Errorf("failed to convert start coordinate %d to integer, %s", start, err)
+		c.String(http.StatusBadRequest, "startCoordinate must be an integer")
+
+		return
+	}
+	end, err := strconv.ParseInt(qEnd, 10, 0)
+	if err != nil {
+		log.Errorf("failed to convert end coordinate %d to integer, %s", end, err)
+		c.String(http.StatusBadRequest, "endCoordinate must be an integer")
+
+		return
+	}
+	if end < start {
+		log.Errorf("endCoordinate=%d must be greater than startCoordinate=%d", end, start)
+		c.String(http.StatusBadRequest, "endCoordinate must be greater than startCoordinate")
+
+		return
+	}
+
+	// Calculate the content length
+	contentLength := fileDetails.DecryptedSize
+	if c.Param("type") == "encrypted" {
+		contentLength = fileDetails.ArchiveSize
+		start, end, err = calculateEncryptedCoords(start, end, c.GetHeader("Range"), fileDetails)
```

Comment:
Feels odd to manage coordinates coming from the header within the `calculateEncryptedCoords` when coordinates coming in query is handled before.

Reply:
For unencrypted data, the start and end are just parsed from the query params (line 171 and below). It's only for the encrypted case that we look at the range header and adapt the coords to the data boundaries. One could of course break out the coord parsing code into a separate function, to make it easier to follow, something like (starting line 170):
```go
	start, end, err := parseCoords(c) // get the coords from the params

	// Calculate the content length
	contentLength := fileDetails.DecryptedSize
	if c.Param("type") == "encrypted" {
                // if encrypted, check the range header + adapt the previously set start and end so that it match boundaries
		start, end, err = calculateEncryptedCoords(start, end, c.GetHeader("Range"), fileDetails)
```

Reply:
Yes, I see that they are managed differently and I would prefer if that wasn't the case.

For bigpicture, we have a strong use case for supporting range request as well.

So I'd prefer the "figuring out what parts the user wants" to be refactored into a separate function. I'm fine with a TODO but think it can be done now as well.

Reply:
I created #749 for this.